core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2580,6 +2581,61 @@ private void loadMetadataIfNotExist(RpcContext rpcContext, LockedInodePath inode
     }
   }
 
+  @Override
+  public void updateMount(AlluxioURI alluxioPath, MountContext context)
+      throws FileAlreadyExistsException, FileDoesNotExistException, InvalidPathException,
+      IOException, AccessControlException {
+    LockingScheme lockingScheme = createLockingScheme(alluxioPath,
+        context.getOptions().getCommonOptions(), LockPattern.WRITE_EDGE);
+    try (RpcContext rpcContext = createRpcContext();
+         LockedInodePath inodePath = mInodeTree
+             .lockInodePath(lockingScheme.getPath(), lockingScheme.getPattern());
+         FileSystemMasterAuditContext auditContext = createAuditContext(
+             "updateMount", alluxioPath, null, inodePath.getParentInodeOrNull())) {
+      try {
+        mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+      } catch (AccessControlException e) {
+        auditContext.setAllowed(false);
+        throw e;
+      }
+      MountInfo mountInfo = mMountTable.getMountTable().get(alluxioPath.getPath());
+      if (mountInfo == null) {
+        throw new InvalidPathException("Failed to update mount properties for "
+            + inodePath.getUri() + ". Please ensure the path is an existing mount point.");
+      }
+      // test the new mount configuration first before unmounting the old ufs
+      UnderFileSystemConfiguration ufsConf = UnderFileSystemConfiguration
+          .defaults(ServerConfiguration.global())
+          .createMountSpecificConf(context.getOptions().getPropertiesMap());
+      String ufsUri = mountInfo.getUfsUri().toString();
+      try (UnderFileSystem ufs = UnderFileSystem.Factory.create(ufsUri, ufsConf)) {
+        // check the root path is valid
+        ufs.getFileStatus(ufsUri);
+      } catch (Throwable t) {
+        throw new InvalidArgumentException("Failed to mount under file system."
+            + " Please ensure the options are valid", t);
+      }
+      unmountInternal(rpcContext, inodePath, false);
+      long mountId = IdUtils.createMountId();
```

Comment:
We modify the mount id without journaling?

Reply:
The only down side I can think of is the interruption of current UFS operations. Since the updateMount request is equivalent to user doing an unmount and a mount immediate after, it will immediately invalidate existing mount id, causing any client operation using the old mount id to fail. I talked to @ns1123 and he is okay with the behavior.

The reason I want to use a different mount Id is for worker to invalidate the cached UFS and get a new updated UFS client. If we use the same mount id then there has to be a new RPC call to inform worker of the update.

Reply:
I see, which services currently cache the mount id? I think the worker does?

Reply:
yep.

Reply:
The work ufs manager maintains a UFS client cache which will reuse the clients of the same id.

Reply:
Will the worker gracefully handle needing the refresh the cache after it fails once with the old mount id?

Reply:
no. The current expectation is that the updateMount is a interruptive command similar to unmount which will cause all existing data access request to fail due to the mount id invalidation.